hooks/useStarmapContentUpdated.ts

```diff
@@ -1,17 +1,41 @@
-import { useEffect } from 'react';
-import Toast from 'awesome-toast-component'
+import Toast from 'awesome-toast-component';
+import { useEffect, useState } from 'react';
+import { useGlobalLoadingState } from './useGlobalLoadingState';
+
+const FIRST_LOAD_KEY_PREFIX = 'starmap:content:updated:first-load:';
+
+export default async function useStarmapUpdateAvailable(): Promise<void> {
+  const globalLoadingState = useGlobalLoadingState();
+  const [updateAvailable, setUpdateAvailable] = useState(false);
 
-export default function useStarmapContentUpdated (): void {
   useEffect(() => {
-    window.document.addEventListener('starmap:content:updated', () => {
-      new Toast('Starmap Content Changed! Please Refresh.', {
-        waitForEvent: true,
-        style: {
-          container: [
-            ['cursor', 'pointer'],
-          ]
-        }
-      })
+    window.document.addEventListener('starmap:content:updated', ({ detail }) => {
+      if (detail.cacheName === 'milestones' || detail.cacheName === 'roadmap') {
+        setUpdateAvailable(true);
+      }
     })
-  }, []);
+  });
+
+  useEffect(() => {
+    const firstLoadKey = `${FIRST_LOAD_KEY_PREFIX}${window.location.pathname}`;
+    // If this is the first time the user is loading the page, don't show the toast.
+    // Using localStorage to persist this value across page refreshes.
+    // Because localStorage is synchronous.
+    const isFirstLoad = localStorage.getItem(firstLoadKey) === null;
+    if (updateAvailable && !globalLoadingState.get()) {
+      if (isFirstLoad) {
+        localStorage.setItem(firstLoadKey, 'true');
```

Comment:
I feel like the "first load" should be reset on app initialization and not persist within localStorage.

This was still functioning oddly for me, showing me a toast notification always (except for in a private window). I unregistered the service worker in my main window and reloaded; it seems to be working properly now.